doc/how_to/migrate/from_streamlit.md

```diff
@@ -0,0 +1,950 @@
+        func = calculation_a
+    else:
+        func = calculation_b
+
+    time_start = time.perf_counter()
+    result = func()
+    time_end = time.perf_counter()
+    yield f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete"""
+
+
+calculation_input = pn.widgets.RadioBoxGroup(name="Calculation", options=["A", "B"])
+run_input = pn.widgets.Button(name="Press to run calculation", icon="caret-right", button_type="primary", sizing_mode="fixed", width=250)
+
+pn.Column(
+    "Which calculation would you like to perform?",
+    calculation_input,
+    pn.bind(calculation_message, calculation_input),
+    run_input,
+    pn.bind(calculation_result, run_input, calculation_input)
+).servable()
+```
+
+We use the `pn.indicators.LoadingSpinner` to indicate the activity. You find the full list of indicators in the [Indicators Section](https://panel.holoviz.org/reference/index.html#indicators) of the [Component Gallery](https://panel.holoviz.org/reference/index.html).
+
+The app looks like
+
+![Panel Multiple Updates Examples](https://user-images.githubusercontent.com/42288570/243399714-04345c38-44a8-4bc0-b111-a8d26f03484b.gif)
+
+#### Panel Multiple Updates Alternative Indicator Example
+
+An alternative to using an *indicator* would be to change the `.disabled` and `.loading` parameters of the `calculation_input` and `run_input`.
+
+```python
+import panel as pn
+import time
+import random
+
+pn.extension(sizing_mode="stretch_width", template="bootstrap")
+
+pn.state.template.param.update(site="Panel", title="Calculation Runner")
+
+def calculation_message(calculation):
+    return f"You chose: {calculation}"
+
+def calculation_a():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+
+def calculation_b():
+    time.sleep(1.5)
+    return random.randint(0, 100)
+
+def calculation_result(running, calculation):
+    if not running:
+        yield "Calculation did not run yet"
+        return # This will break the execution
+
+    calculation_input.disabled=calculation_input.loading=run_input.disabled=run_input.loading=True
+
+    if calculation=="A":
+        func = calculation_a
+    else:
+        func = calculation_b
+
+    time_start = time.perf_counter()
+    result = func()
+    time_end = time.perf_counter()
+    calculation_input.disabled=calculation_input.loading=run_input.disabled=run_input.loading=False
+    yield f"""Done!
+
+Result: {result}
+
+The function took {time_end - time_start:1.1f} seconds to complete"""
+
+
+calculation_input = pn.widgets.RadioBoxGroup(name="Calculation", options=["A", "B"])
+run_input = pn.widgets.Button(name="Press to run calculation", icon="caret-right", button_type="primary", sizing_mode="fixed", width=250)
+
+pn.Column(
+    "Which calculation would you like to perform?",
+    calculation_input,
+    pn.bind(calculation_message, calculation_input),
+    run_input,
+    pn.bind(calculation_result, run_input, calculation_input)
+).servable()
```

Comment:
I think if you activate loading, it's automatically disabled; I also wrap try/finally so in case the code breaks in the middle somewhere, it won't stay loading.

```suggestion
import time
import random
from textwrap import dedent

import panel as pn

pn.extension(sizing_mode="stretch_width", template="bootstrap")
pn.state.template.param.update(site="Panel", title="Calculation Runner")


def notify_choice(calculation):
    return f"You chose: {calculation}"


def calculation_a():
    time.sleep(1.5)
    return random.randint(0, 100)


def calculation_b():
    time.sleep(1.5)
    return random.randint(0, 100)


def run_calculation(running, calculation):
    if not running:
        yield "Calculation did not run yet"
        return  # This will break the execution
    try:
        calculation_input.loading = True
        run_input.loading = True
        yield pn.Row(
            pn.indicators.LoadingSpinner(value=True, width=50, height=50, align="center"),
            "Running... Please wait!",
            align="center",
        )
        if calculation == "A":
            func = calculation_a
        else:
            func = calculation_b
        time_start = time.perf_counter()
        result = func()
        time_end = time.perf_counter()
        yield dedent(
            f"""
            Done!
            Result: {result}
            The function took {time_end - time_start:1.1f} seconds to complete
            """
        )
    finally:
        calculation_input.loading = False
        run_input.loading = False


calculation_input = pn.widgets.RadioBoxGroup(name="Calculation", options=["A", "B"])
run_input = pn.widgets.Button(
    name="Press to run calculation",
    icon="caret-right",
    button_type="primary",
    sizing_mode="fixed",
    width=250,
)
pn.Column(
    "Which calculation would you like to perform?",
    calculation_input,
    pn.bind(notify_choice, calculation_input),
    run_input,
    pn.bind(run_calculation, run_input, calculation_input),
).servable()
```

Reply:
Thanks. I think its a good idea. We have to consider whether we provide a minimum amount of text as you would often see in the Streamlit documentation or we would like to provider longer but more robust code.